File: frontend/src/types/history.ts
import { activityTypeLabels } from "./incidents";

export type HistoryCategory = "break" | "shutdown" | "excavation" | "other_incident";
export type HistorySource = "incident" | "planned_shutdown";

export interface HistoryFilters {
  from?: string;
  to?: string;
  category?: HistoryCategory[];
  location?: string;
  page?: number;
  page_size?: number;
}

export interface HistoryItem {
  id: string;
  source: HistorySource;
  category: HistoryCategory;
  activity_type: HistoryCategory;
  number: string;
  title: string;
  status: string;
  occurred_at: string;
  expected_end_at: string | null;
  locations: string[];
  affected_address_count: number | null;
  href: string;
}

export interface HistoryResponse {
  items: HistoryItem[];
  summary: { total: number; breaks: number; shutdowns: number; excavations: number; other_incidents: number };
  page: number;
  page_size: number;
  total_pages: number;
}

export const historyCategoryLabels: Record<HistoryCategory, string> = activityTypeLabels;

File: backend/app/schemas/history.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel

from app.activity import ActivityType


HistoryCategory = ActivityType
HistorySource = Literal["incident", "planned_shutdown"]


class HistoryItem(BaseModel):
    id: UUID
    source: HistorySource
    category: HistoryCategory
    activity_type: ActivityType
    number: str
    title: str
    status: str
    occurred_at: datetime
    expected_end_at: datetime | None
    locations: list[str]
    affected_address_count: int | None
    href: str


class HistorySummary(BaseModel):
    total: int
    breaks: int
    shutdowns: int
    excavations: int
    other_incidents: int


class HistoryResponse(BaseModel):
    items: list[HistoryItem]
    summary: HistorySummary
    page: int
    page_size: int
    total_pages: int
